mysql-test/include/index_merge1.inc

```diff
@@ -516,7 +516,11 @@ if ($eval_before_analyze)
 {
   eval $eval_before_analyze;
 }
-analyze table t0;
+# We use OPTIMIZE instead of ANALYZE here to handle the case when InnoDB
+# purge threads do not manage to keep up and completely remove records which
+# were marked for deletion earlier, which affects row count estimations in
+# EXPLAIN below.
+optimize table t0;
```

Comment:
this is a bit of cheating :stuck_out_tongue:  optimize table essentially rebuilds the table in innodb.
May be you should use mysql-test/include/wait_innodb_all_purged.inc?

Reply:
Right. If the intention is not to test the execution or behavior of InnoDB, then rebuilding is OK.